build.gradle.kts

```diff
@@ -120,6 +126,12 @@ sourceSets {
         runtimeClasspath += sourceSets["test"].output
         runtimeClasspath += sourceSets["main"].output
     }
+    create("e2eTest") {     //separate infrastructure as compatTest is called multiple times with different Java versions
+        compileClasspath += sourceSets["compatTest"].output
+        compileClasspath += sourceSets["main"].output
+        runtimeClasspath += sourceSets["compatTest"].output
+        runtimeClasspath += sourceSets["main"].output
+    }
```

Comment:
If you create the source set outside of the `sourceSet` "block", you can reference it below.


```kotlin
val e2eTest by sourceSets.creating {
    compileClasspath += sourceSets["compatTest"].output
    compileClasspath += sourceSets["main"].output
    runtimeClasspath += sourceSets["compatTest"].output
    runtimeClasspath += sourceSets["main"].output
}
```